### PR TITLE
Feat: add usage of gravatar for user avatars

### PR DESCRIPTION
**Pull request by Ne-wt**

### CODE REVIEW

**Review by itinerare — APPROVED** [review]

**Review by Draginraptor — APPROVED** [review]

File: app/Models/User/User.php [hunk 1]
```diff
@@ -333,6 +333,28 @@ public function getAvatar()
...
+    public function getAvatarUrlAttribute()
+    {
+        if ($this->avatar == 'default.jpg') {
+            // check if a gravatar exists
+            $hash = md5(strtolower(trim($this->email)));
+            $url = 'https://www.gravatar.com/avatar/'.$hash.'??d=mm&s=200';
+            $headers = @get_headers($url);
+            return $url;
+            if (!preg_match("|200|", $headers[0])) {
+                return url('assets/images/avatars/default.jpeg');
```
Comment (SpeedyD) [inline app/Models/User/User.php:350]: ...Isn't this a problem? Filename is `default.jpeg`, while the default file is `default.jpg`.

**Review by SpeedyD — CHANGES_REQUESTED** [review]
Looks fine except for the jpeg vs jpg file extension..

File: app/Models/User/User.php (hunk 1 above)
Comment (Draginraptor) [reply app/Models/User/User.php:350]: I wanted to look into this more, and also realised that there's also `return $url;` on line 348, which means the if-else wouldn't be reached.

That said, `url('assets/images/avatars/default.jpeg')` should also be `url('images/avatars/default.jpg')` (inclusive of the jpeg to jpg change.

**Review by Draginraptor — CHANGES_REQUESTED** [review]
Check over the comments on the User model's getAvatarUrlAttribute function

**Review by itinerare — APPROVED** [review]

**Review by AW0005 — APPROVED** [review]

**Pull request merged**